src/quoter.ts

```diff
@@ -69,15 +68,15 @@ export abstract class SwapQuoter {
 
     if (singleHop) {
       const quoteV2Params: QuoteParams = {
-        amount: quoteAmount,
-        fee: route.pools[0].fee,
-        sqrtPriceLimitX96: toHex(options?.sqrtPriceLimitX96 ?? 0),
         tokenIn: route.tokenPath[0].address,
         tokenOut: route.tokenPath[1].address,
+        fee: route.pools[0].fee,
+        amount: quoteAmount,
+        sqrtPriceLimitX96: toHex(options?.sqrtPriceLimitX96 ?? 0)
```

Comment:
I didn't realize order matters here. It seems like that detail could easily be lost in future work. Can you add a comment, or better yet manually enforce order in your code?

Reply:
yeah the order matters and Object.values should not be relied upon, +1 to fixing this